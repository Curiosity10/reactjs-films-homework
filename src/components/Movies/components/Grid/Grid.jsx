import React, { useState, useCallback } from 'react';
import GridCard from './components/GridCard';

const Grid = ({ movies }) => {
  const [activeCard, setActiveCard] = useState(0);
  const changeActiveCard = useCallback(
    (id) => {
      setActiveCard(id);
    },
    [],
  );

  return movies.map(movie => (
    <GridCard
      activeCard={activeCard}
      changeActiveCard={changeActiveCard}
      key={movie.id}
      movie={movie}
      isActive={movie.id !== activeCard}
    />
  ));
};

export default Grid;
