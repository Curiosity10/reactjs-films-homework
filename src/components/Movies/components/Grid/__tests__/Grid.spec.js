import React from 'react';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Grid from '..';

describe('Grid component renders correctly', () => {
  const movies = [
    {
      title: 'test',
      id: 1,
      genre_ids: [
        {
          id: 12,
          name: 'Action',
        },
        {
          id: 16,
          name: 'Adventure',
        },
      ],
      overview: 'desc',
    },
    {
      title: 'test2',
      id: 2,
      genre_ids: [
        {
          id: 21,
          name: 'Action',
        },
        {
          id: 23,
          name: 'Adventure',
        },
      ],
      overview: 'desc2',
    },
  ];
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      genreId: null,
      searchQuery: '',
      videoKey: '',
      pagesCount: 1,
    },
    ui: {
      layout: 'grid',
    },
  };
  const store = mockStore(initialState);

  it('Grid renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Change active card called correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    );
    const mockFunction = jest.fn(tree.toTree().rendered.props.changeActiveCard);

    act(() => {
      mockFunction();
    });

    expect(mockFunction).toHaveBeenCalled();
  });

  it('Change active card works correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    );

    const mockFunction = jest.fn(
      id => tree.toTree().rendered.rendered[0].props.changeActiveCard(id),
    );

    act(() => {
      mockFunction(123);
    });

    expect(mockFunction).toHaveBeenCalled();
  });
});
