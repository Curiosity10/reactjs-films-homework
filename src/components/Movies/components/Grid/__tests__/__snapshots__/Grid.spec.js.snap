// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly Grid renders correctly 1`] = `
Array [
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
          title="test"
        >
          <a
            href="/movie/:1"
            onClick={[Function]}
          >
            test
          </a>
        </h3>
        <p />
      </div>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
          title="test2"
        >
          <a
            href="/movie/:2"
            onClick={[Function]}
          >
            test2
          </a>
        </h3>
        <p />
      </div>
    </div>
  </article>,
]
`;
